test/test_prototype_transforms_functional.py

```diff
@@ -1168,25 +1157,67 @@ def _compute_expected_bbox(bbox, padding_):
         torch.testing.assert_close(output_boxes, expected_bboxes)
 
 
+@pytest.mark.parametrize("device", cpu_and_gpu())
+def test_correctness_pad_segmentation_mask_on_fixed_input(device):
+    mask = torch.ones((1, 3, 3), dtype=torch.long, device=device)
+
+    out_mask = F.pad_segmentation_mask(mask, padding=[1, 1, 1, 1])
+
+    expected_mask = torch.zeros((1, 5, 5), dtype=torch.long, device=device)
+    expected_mask[:, 1:-1, 1:-1] = 1
+    torch.testing.assert_close(out_mask, expected_mask)
+
+
 @pytest.mark.parametrize("padding", [[1, 2, 3, 4], [1], 1, [1, 2]])
-def test_correctness_pad_segmentation_mask(padding):
-    def _compute_expected_mask(mask, padding_):
+@pytest.mark.parametrize("padding_mode", ["constant", "edge", "reflect", "symmetric"])
+def test_correctness_pad_segmentation_mask(padding, padding_mode):
+    def _compute_expected_mask(mask, padding_, padding_mode_):
         h, w = mask.shape[-2], mask.shape[-1]
         pad_left, pad_up, pad_right, pad_down = _parse_padding(padding_)
 
+        # expected output can be computed on positive pad values only
+        # but F.pad_* can also crop for negative values
+        assert pad_left > 0 and pad_up > 0 and pad_right > 0 and pad_down > 0
```

Comment:
Let's make it clear that this is not an error of the op.

```suggestion
        if any(pad <= 0 for pad in [pad_left, pad_up, pad_right, pad_down])
            raise pytest.UsageError(
                "expected output can be computed on positive pad values only, "
                "but F.pad_* can also crop for negative values"
            )
```